Nos/Controller/SearchController.swift

```diff
@@ -69,6 +70,10 @@ class SearchController: ObservableObject {
             .filter { !$0.isEmpty }
             .compactMap { [weak self] query in
                 guard let self else { return nil }
+                if self.state == .noQuery {
```

Comment:
Well, if that's true then it would be confusing yeah xd. What do you recommend in order to just send the event once when the user starts searching? In my tests this worked fine.

Reply:
I looked at the code more carefully today and, while you are correct and I understand what you mean when you say that the state ca be .noQuery, I think this is the best solution we have.

I didn't find a clean way of detecting that the user made a new search. At some point, when there was already text in the search field and the user started typing again, it's guessing the intention of the user. Is it the same search that the user is correcting or it is a whole new search? So I conclude that we cannot possibly catch every possible case, just do our best.

With this solution I'm proposing we are detecting correctly searches made from an empty state, searches made after tapping on the clear button, and searches made after selecting the whole text, deleting and starting over. What we are not detecting is searches made after selecting the whole text and, instead of deleting or tapping on the clear button, typing a new search straightaway. While the user could be starting a new search, it might just be correcting himself. I discarded other options like tracking each time the user types something because it would be too many events.

Reply:
I think you're right and this is our best solution for now. Let's go with it and see what happens.